### ISSUE TITLE
Feature Request: Place `ON CONFLICT` on new line

### ISSUE THREAD

**Issue by markhilb** [sql-formatter-org/sql-formatter#535]
**Describe the Feature**
Place `ON CONFLICT` on new line.

Current behaviour:
```sql
INSERT INTO
    tests (val)
VALUES
    (1),
    (2) ON CONFLICT (val)
DO NOTHING;
```

Wanted behaviour:
```sql
INSERT INTO
    tests (val)
VALUES
    (1),
    (2) 
ON CONFLICT (val)
DO NOTHING;
```

**Comment by nene** [comment]
Another problem in here really is that `DO NOTHING` gets placed on a new line. This only happens with MySQL, MariaDB, SingleStoreDB and PostgreSQL (though not sure if `ON CONFLICT` clause is valid syntax in all of them).

**Comment by nene** [comment]
This is now fixed in 12.0.4 release.
